src/raster/r.boxplot/r.boxplot.py

```diff
@@ -0,0 +1,633 @@
+#               for details.
+#
+#############################################################################
+
+# %module
+# % description: Draws the boxplot of raster values. Optionally, this is done per category of a zonal raster layer
+# % keyword: display
+# % keyword: raster
+# % keyword: plot
+# % keyword: boxplot
+# %end
+
+# %option G_OPT_R_MAP
+# % key: input
+# % description: input raster
+# % required: yes
+# % label: input raster
+# % guisection: Input
+# %end
+
+# %option G_OPT_R_MAP
+# % key: zones
+# % label: zonal raster
+# % description: categorical map with zones
+# % required: no
+# % guisection: Input
+# %end
+
+# %option G_OPT_F_OUTPUT
+# % key: output
+# % label: Name of output image file
+# % required: no
+# % guisection: Output
+# %end
+
+# %option
+# % key: order
+# % type: string
+# % label: Sort boxplots
+# % description: Sort boxplots based on their median values
+# % required: no
+# % options: descending,ascending
+# % guisection: Plot options
+# %end
+
+# %option
+# % key: range
+# % type: double
+# % label: Range (value > 0)
+# % description: this determines how far the plot whiskers extend out from the box. If range is positive, the whiskers extend to the most extreme data point which is no more than range times the interquartile range from the box. A value of zero causes the whiskers to extend to the data extremes.
+# % required: no
+# % answer: 1.5
+# % guisection: Plot options
+# %end
+
+# %option
+# % key: map_outliers
+# % type: string
+# % label: Name of outlier map
+# % description: Create a vector point layer of outliers
+# % guisection: Output
+# % required: no
+# %end
+
+# %flag
+# % key: h
+# % label: horizontal boxplot(s)
+# % description: Draw the boxplot horizontal
+# % guisection: Plot options
+# %end
+
+# %flag
+# % key: o
+# % label: include outliers
+# % description: Draw boxplot(s) with outliers
+# % guisection: Plot options
+# %end
+
+# %flag
+# % key: n
+# % label: notch
+# % description: Draw boxplot(s) with notch
+# % guisection: Plot options
+# %end
+
+# %flag
+# % key: r
+# % label: rotate labels
+# % description: rotate labels
+# % guisection: Plot options
+# %end
+
+# %option
+# % key: plot_dimensions
+# % type: string
+# % label: Plot dimensions (width,height)
+# % description: Dimensions (width,height) of the figure in inches
+# % required: no
+# % guisection: Plot options
+# %end
+
+# %option
+# % key: dpi
+# % type: integer
+# % label: DPI
+# % description: resolution of plot
+# % required: no
+# % guisection: Plot options
+# %end
+
+# %rules
+# % requires: map_outliers, -o
+# %end
+
+import sys
+import atexit
+import uuid
+from subprocess import PIPE
+import grass.script as gs
+from grass.pygrass.modules import Module
+
+try:
+    import matplotlib
+
+    matplotlib.use("WXAgg")
+    from matplotlib import pyplot as plt
+except ImportError as e:
+    raise ImportError(
+        _(
+            'r.boxplot  needs the "matplotlib" '
```

Comment:
```suggestion
            'r.boxplot needs the "matplotlib" '
```
This try/except might go as lazy import after calling the parser, right? It was my understanding that lazy import is good practice.

Reply:
> This try/except might go as lazy import after calling the parser, right? It was my understanding that lazy import is good practice.

@veroandreo changed it, can you check?